packages/editor/src/client/atoms/permissionControl.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {ComponentType, PropsWithChildren} from 'react'
+import {AuthContext} from '../authContext'
+import {useTranslation} from 'react-i18next'
+
+interface PermissionControlProps {
+  requiredPermission: string
+  showMessage?: boolean
+}
+
+export function PermissionControl({
+  children,
+  requiredPermission,
+  showMessage
+}: PropsWithChildren<PermissionControlProps>) {
+  const {t} = useTranslation()
+  return (
+    <AuthContext.Consumer>
+      {value => (
+        <>
+          {value.session?.roles?.some(role =>
+            role.permissions.some(permission => permission.id === requiredPermission)
+          ) ? (
+            children
+          ) : showMessage ? (
+            <p>{t('permissions.noAccess')}</p>
```

Comment:
Doesn't look that nice in the UI, when I test it. The sentence seams a bit lost. You may want to add an icon, align the text and choose probably a bigger font size? Maybe even add some more explanatory text?

![image](https://user-images.githubusercontent.com/70939054/184110385-de9cd9f6-ea0b-4ccd-ba87-b7fca2fe2531.png)